src/include/function/gds/bfs_graph.h

```diff
@@ -41,79 +45,138 @@ class ParentList {
     std::atomic<ParentList*> next;
 };
 
-class BFSGraph {
-    friend class BFSGraphInitVertexCompute;
-    // Data type that is allocated to max num nodes per node table.
-    using parent_entry_t = std::atomic<ParentList*>;
+class BaseBFSGraph {
```

Comment:
BFSGraph should also make use of `GSDObjects<ParentList*>` to handle sparse and dense data structures, instead of relying on derived `DenseBFSGraph` and `SparseBFSGraph`.